modules/mod_comment/mod_comment.erl

```diff
@@ -67,23 +69,38 @@ event(#submit{message={newcomment, Args}, form=FormId}, Context) ->
             Html = z_template:render(CommentTemplate, Props, Context),
             Context1 = z_render:insert_bottom(CommentsListElt, Html, Context),
             Context2 = case Is_visible of
-			   true ->
-			       z_render:wire([
-					      {set_value, [{selector, <<"#", FormId/binary, " textarea[name=\"message\"]">>}, {value, <<>>}]},
-					      {set_value, [{selector, <<"#", FormId/binary, " input[name=\"message\"]">>}, {value, <<>>}]},
-					      {fade_in, [{target, "comment-"++integer_to_list(CommentId)}]}
-					     ], Context1);
-			   false ->
-			       Context1
-		       end,
+                           true ->
+                               z_render:wire([
+                                   {set_value, [{selector, <<"#", FormId/binary, " textarea[name=\"message\"]">>}, {value, <<>>}]},
+                                   {set_value, [{selector, <<"#", FormId/binary, " input[name=\"message\"]">>}, {value, <<>>}]},
+                                   {fade_in, [{target, "comment-" ++ integer_to_list(CommentId)}]}
+                               ], Context1);
+                           false ->
+                               Context1
+                       end,
+
             case z_convert:to_bool(proplists:get_value(do_redirect, Args, true)) of
-                true -> z_render:wire({redirect, [{location, "#comment-"++integer_to_list(CommentId)}]}, Context2);
+                true -> z_render:wire({redirect, [{location, "#comment-" ++ integer_to_list(CommentId)}]}, Context2);
                 false -> Context2
             end;
         {error, _} ->
             Context
     end.
 
+is_visible(Context) ->
+    case m_config:get_value(comments, moderate, Context) of
+        <<"1">> -> false;
+        _Else -> true
+    end.
+
+get_rating(Context) ->
+    case z_context:get_q(<<"rating">>, Context) of
+        undefined -> undefined;
+        RatingString ->
+            Rating = z_convert:to_integer(RatingString),
+            true = 0 =< Rating andalso Rating =< 5,
+            Rating
+     end.
 
 %% @doc Return the list of recent comments.  Returned values are the complete records.
 observe_search_query(#search_query{search={recent_comments, []}, offsetlimit=OffsetLimit}, Context) ->
```

Comment:
Line 106 is too long: observe_search_query(#search_query{search={recent_comments, []}, offsetlimit=OffsetLimit}, Context) ->.